consensus/module.go

```diff
@@ -109,6 +112,38 @@ func (m *consensusModule) SetUtilityContext(utilityContext modules.UtilityContex
 	m.utilityContext = utilityContext
 }
 
+// Implementations of the ConsensusStateSync interface
+func (m *consensusModule) GetLeaderId() uint64 {
+	if m.leaderId == nil {
+		//REFACTOR: As per issue #434, once new id is sorted out, this return statement must be changed
+		return 0
+	}
+	return uint64(*m.leaderId)
+}
+
+func (m *consensusModule) GetNodeIdFromNodeAddress(peerId string) (uint64, error) {
+
+	validators, err := m.getValidatorsAtHeight(m.CurrentHeight())
+	if err != nil {
+		//REFACTOR: As per issue #434, once new id is sorted out, this return statement must be changed
+		return 0, err
+	}
+
+	valAddrToIdMap := typesCons.NewActorMapper(validators).GetValAddrToIdMap()
+	return uint64(valAddrToIdMap[peerId]), nil
+}
+
+func (m *consensusModule) GetCurrentNodeAddressFromNodeId() (string, error) {
+	validators, err := m.getValidatorsAtHeight(m.CurrentHeight())
+	if err != nil {
+		return "", err
+	}
+
+	idToValAddrMap := typesCons.NewActorMapper(validators).GetIdToValAddrMap()
+
+	return idToValAddrMap[typesCons.NodeId(m.nodeId)], nil
```

Comment:
no, because this is supposed to return NodeAddress given id using `IdToValAddrMap`, rather `GetNodeIdFromNodeAddress` returns node id using `ValAddrToIdMap`.